clippy_lints/src/let_underscore.rs

```diff
@@ -88,6 +112,16 @@ impl<'tcx> LateLintPass<'tcx> for LetUnderscore {
                     "consider using an underscore-prefixed named \
                             binding or dropping explicitly with `std::mem::drop`",
                 );
+            } else if let Some(future_trait_def_id) = cx.tcx.lang_items().future_trait()
+                && implements_trait(cx, cx.typeck_results().expr_ty(init), future_trait_def_id, &[]) {
+                span_lint_and_help(
+                    cx,
+                    LET_UNDERSCORE_FUTURE,
+                    local.span,
+                    "non-binding let on a future",
```

Comment:
I'd suggest putting the keyword in backticks, as in:
```
"non-binding `let` on a future",
```

Reply:
Good call, should I adjust the other lint messages in this file as well?

Reply:
Yes, that makes sense.

Reply:
Gotcha, I figured as much so I did that in my changed commits